producer.go

```diff
@@ -535,16 +535,25 @@ func (p *producer) dispatchWork(workCtx context.Context, count int, fetchResultC
 func (p *producer) heartbeatLogLoop(ctx context.Context) {
 	ticker := time.NewTicker(5 * time.Second)
 	defer ticker.Stop()
+	type state struct {
```

Comment:
I  think the naming on this is just a little too generic. Maybe `jobStatistics`?

Reply:
It's not really "statistics", right? At least, not how I would use the word: It's the state at the time this is run. I changed it to `jobState`, but if you want `jobStatistics` then I can change it to that.

Reply:
I'm open to alternatives, but the new log message is definitely wrong. Try to zoom out and re-read "producer job state changed" — that sounds like the state of a job is changing. "State" is a common word used in the API for jobs indicating their currently state. e.g. "available", "completed", etc.

Maybe "job counts" or something?

Reply:
> Maybe "job counts" or something?

Yes, that's better :+1: